sklearn/manifold/tests/test_t_sne.py

```diff
@@ -872,13 +884,64 @@ def test_tsne_with_different_distance_metrics():
     for metric, dist_func in zip(metrics, dist_funcs):
         X_transformed_tsne = TSNE(
             metric=metric, n_components=n_components_embedding,
-            random_state=0, n_iter=300).fit_transform(X)
+            random_state=0, n_iter=300, square_distance=True).fit_transform(X)
         X_transformed_tsne_precomputed = TSNE(
             metric='precomputed', n_components=n_components_embedding,
-            random_state=0, n_iter=300).fit_transform(dist_func(X))
+            random_state=0, n_iter=300,
+            square_distance=True).fit_transform(dist_func(X))
         assert_array_equal(X_transformed_tsne, X_transformed_tsne_precomputed)
 
 
+@pytest.mark.parametrize('method', ['exact', 'barnes_hut'])
+@pytest.mark.parametrize('metric', ['euclidean', 'manhattan'])
+@pytest.mark.parametrize('square_distance', [True, 'legacy'])
+@ignore_warnings(category=FutureWarning)
+def test_tsne_with_different_square_distances(method, square_distance, metric):
+    """Make sure that TSNE works for different square_distance settings"""
```

Comment:
Nit: we use comments instead of docstrings for tests

```suggestion
	# Make sure that TSNE works for different square_distance settings
```

Reply:
I was unsure about this myself, so thank you for the clarification.